### DISCUSSION TITLE
Jupyter notebook based pipelining using DVC?

### DISCUSSION THREAD

**Discussion by kurianbenoy** [treeverse/dvc#5930] [New Features & Ideas]
How can DVC be used in a jupyter notebook based system for pipelining each stages? Where assuming all stages for data processing, feature engineering to even model tuning. We use just one single jupyter notebook system?

**Comment by shcheklein** [comment]
from the top of my head - system with Python decorators @danfischetti implemented for his team comes to my mind as a solution for this (it's not publicly available yet).

**Comment by casperdcl** [comment]
NB would probably be useful to change `from tqdm import tqdm` to `from tqdm.auto import tqdm` to get nice notebook progress bars

**Comment by efiop** [comment]
@casperdcl Oh, it will make it detect if it is running from a notebook? Neat!

**Comment by casperdcl** [comment]
Yes
